src/services/ai_service.py

```diff
@@ -1,66 +1,89 @@
-import json
-from typing import List, Dict, Any, Optional
-import google.generativeai as Client
+from typing import List, Dict
 from unidiff import Hunk, PatchedFile
 from ..core.config import Config
-from ..core.models import PRDetails, FileInfo
-
+from ..core.models import PRDetails
+from .llms.gemini import GeminiService
+from .llms.openai import OpenAIService
 
 class AIService:
-  def __init__(self):
-    self.model = Client.GenerativeModel(Config.GEMINI_MODEL)
-
-  def create_prompt(
-    self, file: PatchedFile, hunk: Hunk, pr_details: PRDetails
-  ) -> str:
-    return f"""Your task is reviewing pull requests. Instructions:
-    - Provide the response in following JSON format: {{"reviews": [{{"lineNumber": <line_number>, "reviewComment": "<review comment>"}}]}}
-    - Provide comments and suggestions ONLY if there is something to improve, otherwise "reviews" should be an empty array.
-    - Use GitHub Markdown in comments
-    - Focus on bugs, security issues, and performance problems
-    - IMPORTANT: NEVER suggest adding comments to the code
-
-    Review the following code diff in the file "{file.path}" and take the pull request title and description into account.
-    Pull request title: {pr_details.title}
-    Pull request description:
+    """
+    Service manager that handles selection and usage of available LLM services.
+    Prioritizes Gemini if both services are available.
+    """
+    
+    def __init__(self):
+        """Initialize available LLM services based on configuration."""
+        self.active_service = None
+        self._initialize_service()
 
-    ---
-    {pr_details.description or 'No description provided'}
-    ---
+    def _initialize_service(self) -> None:
+        """
+        Initialize the appropriate LLM service based on available API keys.
+        Prioritizes Gemini over OpenAI if both are available.
+        """
+        try:
+            if hasattr(Config, 'GEMINI_API_KEY') and Config.GEMINI_API_KEY:
+                self.active_service = GeminiService()
+                print("Initialized Gemini service")
+                return
+        except Exception as e:
+            print(f"Failed to initialize Gemini service: {e}")
 
-    Git diff to review:
+        try:
+            if hasattr(Config, 'OPENAI_API_KEY') and Config.OPENAI_API_KEY:
```

Comment:
The `get_ai_response` function catches a generic `Exception`.  Consider catching more specific exceptions (e.g., API-related exceptions) for better error handling and more informative logging.  This will improve the ability to diagnose and address issues.